materials/tidyr.md

```diff
@@ -0,0 +1,112 @@
+---
+layout: page
+element: notes
+title: tidyr
+language: SQL
+---
+
+> Set up R console:
+
+```
+library(tidyr)
+```
+
+### Remember the five basic rules of database structure
+
+1. Order doesn’t matter 
+2. No duplicate rows
+3. Every cell contains one value
+4. One column per type of information
+5. No redundant information
+
+### Restructure tables with messy data 
+
+* Cells with multiple values breaks rule #3.
+* Redundant column information or cross-tabulated data breaks rule #4.
+
+![How to restructure to keep no duplicate rows and one value per cell]({{ site.baseurl }}/materials/database_struct_multiple_habitat_values.png)
+
+* Here is another messy dataset.
+
+```
+scary_sightings <- data.frame(
+  animals = c("lions", "tigers", "bears"),
+  brick_road = c("1-Y", "0-N", "0-N"),
+  emerald_city = c("17-N", "8-Y", "64-N")
+)
+```
+
+```
+> scary_sightings
+  animals brick_road emerald_city
+1   lions        1-Y         17-N
+2  tigers        0-N          8-Y
+3   bears        0-N         64-N
+```
+
+> Ask students,
+> 
+> * "What makes `scary_sightings` messy?"
+> * "What are the variables in `scary_sightings`?"
+
+* Tidy variables in `scary_sightings`
+    * `animals` 
+        * `lions` and `tigers` and `bears`
+    * `site` 
+        * `brick_road` and `emerald_city`
+    * `sightings`
+        * count
+    * `scared`
+        * `Y` or `N`
+
+### `tidyr` helps restructure messy data
+
+* `gather()`
+    * Removes redundant columns
+    * Arguments:
+        * Piped `data.frame`
+        * Column names
+        * Column range
```

Comment:
I think we should be a little clearer here about what the column names and range are and separate the column name for the column that holds the headers from the one that holds the values.